src/Compilers/Core/Portable/ConstantValueSpecialized.cs

```diff
@@ -560,6 +560,22 @@ public override sbyte SByteValue
                 }
             }
 
+            public override short Int16Value
```

Comment:
No, unsigned operations do not need special casing. 
The underlying field is unsigned, so it will be properly zero-extended. (returning 255 for byte.MaxValue)


---
In reply to: [110446716](https://github.com/dotnet/roslyn/pull/18521#discussion_r110446716) [](ancestors = 110446716)